### PR TITLE
docs: Permission request UI cookbook

### PR DESCRIPTION
**Pull request by szuperaz**

### CODE REVIEW

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx [hunk 1]
```diff
@@ -3,4 +3,368 @@ title: Permission Requests
...
+
+- [Register for a Stream account](https://getstream.io/try-for-free/) and obtain our API key and secret.
+- Install the Stream React Video SDK:
+  - `npm install @stream-io/video-react-sdk`
+  - `yarn add @stream-io/video-react-sdk`
+- [Initialize the SDK](../../guides/client-auth) by passing in your API key, token and user information:
+
+:::note
+For the token generation, you can use our [Token Generator](https://getstream.io/chat/docs/token_generator/).
+:::
```
Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx:29]: I like this section/content. I think we should use it across all chapters 💯

**Review by oliverlaz — APPROVED** [review]

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx [hunk 2]
```diff
@@ -3,4 +3,368 @@ title: Permission Requests
...
+};
+
+const PermissionRequests = () => {};
+
+const PermissionRequestNotifications = () => {};
+```
+
+:::note
+
+Permission requests are only relevant for [call types](../../guides/10-configuring-call-types/) that have roles where a participant can't publish their audio/video/screen-share by default.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx:92]: Would it be contra productive to enumerate those call types in this note?

Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx:92]: Reading the referenced doc, it was not immediately clear to me where is the dividing line in this aspect among the call types.

Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx:92]: To be honest I don't know what those call types are, but yeah, we should include that information in the docs one we know them

File: packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx [hunk 3]
```diff
@@ -3,4 +3,368 @@ title: Permission Requests
...
+
+  const shouldDisplayButton = (capability) => {
+    return (
+      call.permissionsContext.canRequest(capability) &&
+      !call.permissionsContext.hasPermission(capability)
+    );
+  };
+
+  const requestPermission = (capability) => {
+    call.requestPermissions({ permissions: [capability] });
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx:131]: Permissions and capabilities are synonyms?

Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx:131]: Yes, they are called `permission` in one part of the API, and `OwnCapability` in the other

**Review by MartinCupela — APPROVED** [review]

**Pull request merged**